### ISSUE TITLE
Can't include or use PouchDB

### ISSUE THREAD

**Issue by episage** [mxstbr/react-boilerplate#1007]
When I `npm i pouchdb-browser --save`
then in `app.js`:

``` js
import PouchDB from 'pouchdb-browser';
```

then webpack throws:

```
ERROR in ./~/debug/node.js
Module not found: Error: Can't resolve 'fs' in '/Users/epi/Projects/timecamp/v3.0/frontend/node_modules/debug'
 @ ./~/debug/node.js 163:15-28

ERROR in ./~/debug/node.js
Module not found: Error: Can't resolve 'net' in '/Users/epi/Projects/timecamp/v3.0/frontend/node_modules/debug'
 @ ./~/debug/node.js 170:16-30
```

![image](https://cloud.githubusercontent.com/assets/2196095/18703374/ef318688-7fe5-11e6-9e0a-45577f69679d.png)

I already tried (webpack.dev.babel.js):

```
  node: {
    fs: 'empty',
    child_process: 'empty',
    net: 'empty',
    tls: 'empty',
  },
```

but it does not help, same errors.

Could you please help me include PouchDB?

Best
episage

**Comment by amilajack** [comment]
My best guess is that you need to add a polyfill, say [node-libs-browser](https://github.com/webpack/node-libs-browser)?

**Comment by Proxiweb** [comment]
"Can't resolve 'fs'" ... the las time I saw this message, I had to upgrade npm to the latest version.

**Comment by episage** [comment]
@amilajack @Proxiweb Sorry guys. It doesn't work:

![image](https://cloud.githubusercontent.com/assets/2196095/19222362/65dc9950-8e4e-11e6-855d-827abd4fc899.png)


![image](https://cloud.githubusercontent.com/assets/2196095/19222359/5bea480c-8e4e-11e6-9045-113f4eb59ca1.png)


![image](https://cloud.githubusercontent.com/assets/2196095/19222357/4ebd958a-8e4e-11e6-9461-16885c8c6df5.png)

![image](https://cloud.githubusercontent.com/assets/2196095/19222366/7a189ab8-8e4e-11e6-9016-fd92a5c508ba.png)

**Comment by amilajack** [comment]
@episage the latest node version is 6.7.0 and latest npm is 3.10.8.

**Comment by amilajack** [comment]
Try this https://github.com/pouchdb/pouchdb/issues/3319#issuecomment-68448181

**Comment by episage** [comment]
@amilajack Upgraded both to the latest versions: 
node is 6.7.0 
npm 3.10.8

Same story. I tried to:
```
  node: {
    console: true,
    fs: 'empty',
    child_process: 'empty',
    net: 'empty',
    tls: 'empty',
  },
```

in `webpack.dev.babel` but it doesn't help either. Which is weird a bit because it seems like the boilerplate doesn't pick up the `node` property.

![image](https://cloud.githubusercontent.com/assets/2196095/19231352/78977a1e-8ed1-11e6-8481-1c9bf3a2f517.png)


I couldn't add it to `webpack.base.babel` because it throws something like this in browser despite correct syntax in the file:

![image](https://cloud.githubusercontent.com/assets/2196095/19231399/be8c4536-8ed1-11e6-801b-4528e8dd9d58.png)


![image](https://cloud.githubusercontent.com/assets/2196095/19231417/cd2cbed6-8ed1-11e6-9087-25a28c05cf48.png)

**Comment by amilajack** [comment]
@episage have you made any modifications to the boilerplate?

**Comment by episage** [comment]
@amilajack I just installed a fresh boilerplate via `git clone ...` `npm run setup` `npm run clean` and run `npm start`. The results are exactly the same, but error log is more detailed:
I added 
```
node: {
  fs: 'empty',
  net: 'empty',
},
```
`webpack.dev.babel.js`

![image](https://cloud.githubusercontent.com/assets/2196095/19265907/2c589196-8f9f-11e6-973e-21010b181f1f.png)

**Comment by episage** [comment]
Funny thing. `debug` package.json has `browser: browser.js` and `main: node.js`. `browser.js` does not import `fs` nor `net`. This is the problem. Webpack`?` doesn't pick up `browser` instead of `main`. Huh. Never heard of `browser` property in `package.json`. Any ideas on how to fix that?

**Comment by mxstbr** [comment]
That's an issue with our webpack setup, we can make webpack recognize the `browser` field by setting that in the resolve option. Can you add it inbetween `jsnext:main` and `main`? Thanks!

**Comment by episage** [comment]
It's already there. Funny how quickly this thing changes... @mxstbr

**Comment by mxstbr** [comment]
Hehe sorry @episage, constantly improving!

**Comment by aminimalanimal** [comment]
@mxstbr
> That's an issue with our webpack setup, we can make webpack recognize the browser field by setting that in the resolve option. Can you add it inbetween `jsnext:main` and `main`?

@episage
> It's already there.

Actually, I just downloaded the current version, and whatever snippet you're referring to doesn't seem to be there.

webpack.base.babel.js:
```
resolve: {
    modules: ['app', 'node_modules'],
    extensions: [
      '',
      '.js',
      '.jsx',
      '.react.js',
    ],
    mainFields: [
      'jsnext:main',
      'main',
    ],
  },
```

This is the only location in the project where I could find `jsnext:main`, so I assume this is where @mxstbr said it should be placed. Perhaps @episage added it locally while debugging this? Or maybe it's changed in the last 12 days?

In any case, I'm still unclear as to how to resolve this issue.

**Comment by episage** [comment]
Check commit No 3515b78bbe2e6102774a21c3e787fde995b82f94 

https://github.com/mxstbr/react-boilerplate/blob/3515b78bbe2e6102774a21c3e787fde995b82f94/internals/webpack/webpack.base.babel.js

Spot on regression :O @aminimalanimal @mxstbr

**Comment by episage** [comment]
Here you are https://github.com/mxstbr/react-boilerplate/pull/1161

**Comment by mxstbr** [comment]
Fixed by the @episage in #1161! Thanks!
